packages/flutter_tools/lib/src/test/test_compiler.dart

```diff
@@ -149,13 +173,19 @@ class TestCompiler {
       } else {
         final File outputFile = fs.file(outputPath);
         final File kernelReadyToRun = await outputFile.copy('${request.path}.dill');
-        final File testCache = fs.file(testFilePath);
-        if (firstCompile || !testCache.existsSync() || (testCache.lengthSync() < outputFile.lengthSync())) {
+        if (firstCompile || testWithLockSize < 0 || testWithLockSize < outputFile.lengthSync()) {
           // The idea is to keep the cache file up-to-date and include as
           // much as possible in an effort to re-use as many packages as
-          // possible.
-          ensureDirectoryExists(testFilePath);
-          await outputFile.copy(testFilePath);
+          // possible. Notice the lock.
```

Comment:
Add a little more context on lock file, Notice the lock that is used to prevent different test processes from reading/writing the file at the same time - or similar.

Reply:
Done.